src/actions/route_helpers/filters.cr

```diff
@@ -0,0 +1,52 @@
+# A series of filters, which may be parsed from the current context. They
+# currently allow filtering the #pkg_file attributes of DPPM::Prefix's Pkg,
+# App, and Src.
+class DppmRestApi::Actions::RouteHelpers::Filters < Hash(String, Set(String))
```

Comment:
This is considered a bad practise, see https://github.com/crystal-lang/shards/pull/230.
Use composition instead.
Just having an internal ivar Hash will perfectly do the job.

Reply:
> Inheriting from stdlib classes

What??? This seems like some incredibly backwards thinking, especially for an OOP language. Using composition is a minor refactor, so I'll do it, but this makes very little sense to me.

Reply:
In any way this shouldn't be a Hash, this is an implementation detail.
In my opinion the Filter type should be a type of its own.

We want the use to use it as a Filter, never as another Hash.
We will also be free to change it to any other type if optimization is needed.

Reply:
In any way this shouldn't be a Hash, this is an implementation detail.
In my opinion the Filter type should be a type of its own.

We want the user to use it as a Filter, never as another Hash.
We will also be free to change it to any other type if optimization is needed.

Reply:
Fair points.